src/app/components/subjects/adding-subject-page/adding-subject/adding-subject.component.ts

```diff
@@ -0,0 +1,143 @@
+
+  subjectInfo: string = "";
+  cabinetInfo: string = "";
+
+  subjectRegExp: any = /^[a-zA-Z/\s]+$/;
+  cabinetRegExp: any = /^[0-9]+$/;
+  firstLetterUppercaseRegExp: any = /^[A-Z]/;
+
+  teachersAll: Teacher[] = [];
+  buttonInfo: string = "Back to subject list";
+
+  messageForSubject: string[] = INFO_MESSAGE_FOR_SUBJECT;
+  messageForCabinet: string[] = INFO_MESSAGE_FOR_CABINET;
+
+  constructor(private dataService: DataService, private router: Router) { }
+
+  ngOnInit(): void {
+    this.getTeachers();
+  }
+
+  getTeachers(): void {
+    this.teachersAll = this.dataService.getDataTeachers();
+  }
+
+  changeItemValue(valueItem: any, itemName: string) {
+    if (itemName === "subject") {
+      this.subject = valueItem;
+      this.checkSubjectCorrectness(valueItem);
+    }
+    if (itemName === "cabinet") {
+      this.cabinet = valueItem;
+      this.checkCabinetCorrectness(valueItem);
+    }
+    if (itemName === "description") {
+      this.description = valueItem;
+    }
+  }
+
+  checkSubjectCorrectness(valueItem: any): void {
+    if (!this.subjectRegExp.test(valueItem) && valueItem) {
+      this.subjectInfo = this.messageForSubject[0];
+      this.subjectCorrectness = false;
+    } else if (!this.firstLetterUppercaseRegExp.test(valueItem) && valueItem) {
+      this.subjectInfo = this.messageForSubject[1];
+      this.subjectCorrectness = false;
+    } else if (this.subjectRegExp.test(valueItem)
+      && this.firstLetterUppercaseRegExp.test(valueItem)) {
+      this.subjectInfo = "";
+      this.subjectCorrectness = true;
+    }
+  }
+
+  checkCabinetCorrectness(valueItem: any): void {
+    if (!this.cabinetRegExp.test(valueItem) && valueItem) {
+      this.cabinetInfo = this.messageForCabinet[0];
+      this.cabinetCorrectness = false;
+    } else if (this.cabinetRegExp.test(valueItem)) {
+      this.cabinetInfo = "";
+      this.cabinetCorrectness = true;
+    }
+  }
+
+  checkSubjectLengthCondition(subject: string): boolean {
+    if (!subject) {
+      this.subjectInfo = this.messageForSubject[4];
+      return false;
+    } else if (subject.length < 4) {
+      this.subjectInfo = this.messageForSubject[2];
+      return false;
+    } else if (subject.length > 15) {
+      this.subjectInfo = this.messageForSubject[3];
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  checkCabinetLengthCondition(cabinet: number): boolean {
+    if (!cabinet) {
+      this.cabinetInfo = this.messageForCabinet[3];
+      return false;
+    } else if (cabinet < 1) {
+      this.cabinetInfo = this.messageForCabinet[1];
+      return false;
+    } else if (cabinet > 20) {
+      this.cabinetInfo = this.messageForCabinet[2];
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  addNewSubject(
+    subject: string,
+    cabinet: number,
+    teachersID: string[],
+    description: string
+  ): void {
+    let descriptionNew: string;
+    description ? descriptionNew = description : descriptionNew = "";
```

Comment:
according to the types, the description can only be a string. Is it necessary to check it?